packages/usa-in-page-navigation/src/index.js

```diff
@@ -188,14 +242,20 @@ const createInPageNav = (inPageNavEl) => {
   const inPageNavContentSelector = Sanitizer.escapeHTML`${
     inPageNavEl.dataset.mainContentSelector || MAIN_ELEMENT
   }`;
+  const inPageNavHeadingSelector = Sanitizer.escapeHTML`${
+    inPageNavEl.dataset.headingElements || IN_PAGE_NAV_HEADINGS
+  }`;
 
   const options = {
     root: null,
     rootMargin: inPageNavRootMargin,
     threshold: [inPageNavThreshold],
   };
 
-  const sectionHeadings = getSectionHeadings(inPageNavContentSelector);
+  const sectionHeadings = getVisibleSectionHeadings(
+    inPageNavContentSelector,
+    inPageNavHeadingSelector
+  );
   const inPageNav = document.createElement("nav");
```

Comment:
Should we have a fallback or message if there are zero sectionHeadings? Consider a unit test to check for empty in-page nav. Pseudocode example below.

**Example**
```js
if (sectionHeadsings.length) {
 // Generate nav.
} else {
	// There aren't any matching visible headings.
	// Consider:
	// 1. Default to all headings.
	// 2. Don't create nav and show message that there aren't matching headers.
}
```

---

## Steps to reproduce

1. Go to `usa-in-page-navigation.twig`
2. In line 11 set change to only use h4's `data-heading-elements="h4"`
3. Visit test url `http://localhost:6006/iframe.html?args=&id=components-in-page-navigation--default&viewMode=story`

<img width="1000" alt="image" src="https://github.com/uswds/uswds/assets/3385219/dded69e3-cc27-4d26-a0aa-9883cad2ec1a">

Reply:
This issue is captured in #5610.